### PR TITLE
Add Sentry to Coverstore

### LINKED ISSUE #3830: Sentry should include the url in its event log
Currently, there's no way to see what URL a user was on when an error was thrown. Let's see if we can add that in.

### Proposal & Constraints
See the sentry docs.

### LINKED ISSUE #4053: Add Sentry to coverstore code
coverstore code should log fixes to Sentry

### Stakeholders
@cclauss

### PR DESCRIPTION
**Pull request by cdrini**
Closes #4053
(Mostly) closes #3830 

Feature. Add sentry logging to covers.

### Technical

Extracted shared logic into some utils, so adding to infobase shouldn't be very difficult.

### Testing
Tested:

- ✅ Locally enabling sentry for covers on py3 + introduce a `1/0`; event logged with request info
- ✅* Locally enabling sentry for covers on py3 + introduce a handled `1/0`; event logged (but _without_ request info)
    - Request info missing; but can fix in a future PR
- ❌ Locally enabling sentry for covers on py2 + introduce a `1/0`; event logged with request info
    - It wasn't logged at all :( Not sure why
- ✅ Locally enabling sentry and try web (py2) node with error; error gets logged.
- ✅ Test on staging.openlibrary.org that errors logged

### Stakeholders
@cclauss @mekarpeles

### CODE REVIEW

**Review by cclauss — APPROVED** [review]
RSLGTM

File: openlibrary/utils/sentry.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    >>> header_name_from_env('CONTENT_LENGTH')
+    'content-length'
+    """
+    header_name = env_name
+    if env_name.startswith('HTTP_'):
+        header_name = env_name[5:]
+    return header_name.lower().replace('_', '-')
+
+
+def add_web_ctx_to_event(event, hint):
```
Comment (cclauss) [inline openlibrary/utils/sentry.py:27]: This is a superpower!!

File: openlibrary/plugins/openlibrary/code.py [hunk 2]
```diff
@@ -37,7 +36,7 @@
 import openlibrary.core.stats
 from openlibrary.plugins.openlibrary.home import format_work_data
 from openlibrary.plugins.openlibrary.stats import increment_error_count  # noqa: E402
-from openlibrary.plugins.openlibrary import processors, sentry
+from openlibrary.plugins.openlibrary import processors
```
Comment (cclauss) [inline openlibrary/plugins/openlibrary/code.py:39]: Why this change?  It is usually better to do imports at the top of the file unless it prevents circular imports.

File: openlibrary/plugins/openlibrary/sentry.py [hunk 3]
```diff
@@ -1,26 +1,14 @@
...
+from openlibrary.utils.sentry import Sentry
 
-from openlibrary.plugins.openlibrary.status import get_software_version
-
-logger = logging.getLogger("openlibrary.sentry")
-
-
-def is_enabled():
-    return hasattr(infogami.config, 'sentry') and infogami.config.sentry.enabled
+sentry = None  # type: Sentry
```
Comment (cclauss) [inline openlibrary/plugins/openlibrary/sentry.py:5]: Not for this PR but for when we finally upgrade to type hints, I believe this will become:
```
from typing import Optional
sentry = None  # type: Optional[Sentry]
```
https://docs.python.org/3/library/typing.html#typing.Optional

**Review by cclauss — APPROVED** [review]
LGTM

File: openlibrary/utils/sentry.py (hunk 1 above)
Comment (cdrini) [reply openlibrary/utils/sentry.py:27]: I haven't been able to get it to work for handled events in coverstore, but it's showing up for a lot of things! 🎉

File: openlibrary/plugins/openlibrary/code.py (hunk 2 above)
Comment (cdrini) [reply openlibrary/plugins/openlibrary/code.py:39]: Ugh; if I import it here, it was `None` :/ Not sure what the best way to make that happen was.
